src/components/Button/__test__/Button.spec.tsx

```diff
@@ -1,92 +1,109 @@
-    expect(screen.getByRole('button', { name: /test button/i })).toBeDisabled();
+    expect(screen.getByRole("button", { name: /test button/i })).toBeDisabled();
   });
 
-  it('disables the button when "isLoading" prop is true', () => {
+  it("disables the button when isLoading prop is true", () => {
     render(<Button isLoading>Test Button</Button>);
-    expect(screen.getByRole('button', { name: /test button/i })).toBeDisabled();
+    expect(screen.getByRole("button", { name: /test button/i })).toBeDisabled();
   });
 
-  it('applies the correct size class based on the "size" prop', () => {
+  it("applies the correct size class based on the size prop", () => {
     const { container } = render(<Button size="lg">Test Button</Button>);
-    expect(container.firstChild).toHaveClass('deriv-button__size--lg');
+    expect(container.firstChild).toHaveClass("deriv-button__size--lg");
   });
 
-  it('applies the correct rounded class based on the "rounded" prop', () => {
+  it("applies the correct rounded class based on the rounded prop", () => {
     const { container } = render(<Button rounded="md">Test Button</Button>);
-    expect(container.firstChild).toHaveClass('deriv-button__rounded--md');
+    expect(container.firstChild).toHaveClass("deriv-button__rounded--md");
   });
 
-  it('shows the icon when provided and not loading', () => {
+  it("shows the icon when provided and not loading", () => {
     const Icon = () => <span>Icon</span>;
     render(<Button icon={<Icon />}>Test Button</Button>);
-    expect(screen.getByText('Icon')).toBeInTheDocument();
+    expect(screen.getByText("Icon")).toBeInTheDocument();
   });
 
-  it('does not show the icon when "isLoading" prop is true', () => {
+  it("does not show the icon when isLoading prop is true", () => {
     const Icon = () => <span>Icon</span>;
     render(<Button isLoading icon={<Icon />}>Test Button</Button>);
-    expect(screen.queryByText('Icon')).toBeNull();
+    expect(screen.queryByText("Icon")).toBeNull();
   });
 
-  it('renders children text when not loading', () => {
+  it("renders children text when not loading", () => {
     render(<Button>Test Button</Button>);
-    expect(screen.getByRole('button', { name: /test button/i })).toHaveTextContent('Test Button');
+    expect(screen.getByRole("button", { name: /test button/i })).toHaveTextContent("Test Button");
   });
 
-  it('does not render children text when "isLoading" prop is true', () => {
+  it("does not render children text when isLoading prop is true", () => {
     render(<Button isLoading>Test Button</Button>);
-    expect(screen.queryByText('Test Button')).toBeNull();
+    expect(screen.queryByText("Test Button")).toBeNull();
   });
+
+  it("hover styles are applied when hasHoverStyle is true", () => {
+    render(
+      <Button hasHoverStyle={true}>Hover Button</Button>
+    );
+    const button = screen.getByRole("button");
+    expect(button).not.toHaveClass("deriv-button__disable-hover");
+  });
```

Comment:
since its defaulted to true, you don't need to pass this prop right? could you check 
```suggestion
  it("hover styles are applied when hasHoverStyle is true", () => {
    render(
      <Button>Hover Button</Button>
    );
    const button = screen.getByRole("button");
    expect(button).not.toHaveClass("deriv-button__disable-hover");
  });
```